src/token/mod.rs

```diff
@@ -348,12 +348,26 @@ fn partial_tokens_to_tokens(mut tokens: &[PartialToken]) -> EvalexprResult<Vec<T
                 },
                 PartialToken::Literal(literal) => {
                     cutoff = 1;
-                    if let Ok(number) = parse_dec_or_hex(&literal) {
+                    let starts_with_alphabet_or_underscore =
+                        literal.starts_with(|x: char| x.is_alphabetic() || x == '_');
+                    // Alphanumeric || Underscore || Colon
+                    let contains_only_valid_chars = literal
+                        .chars()
+                        .all(|x| x.is_alphanumeric() || x == '_' || x == ':');
+                    let is_not_underscore = literal != "_";
+                    let contains_alphabet = literal.contains(|x: char| x.is_alphabetic());
+                    if let Ok(boolean) = literal.parse::<bool>() {
+                        Some(Token::Boolean(boolean))
+                    } else if starts_with_alphabet_or_underscore
```

Comment:
The decomposition of the conditions into separate variables is great!
Here we should first check only `starts_with_alphabet_or_underscore`, and if it does but any of the other three are false, then return a the error variant `IllegalIdentifierSequence`.

Reply:
I mean that the branch after boolean should only check for starts_with_alphabet_or_underscore, and then should have an inner branch that checks the rest.